cs/src/core/Transcoder.cs

```diff
@@ -164,20 +165,25 @@ virtual public void Transcode(R reader, W writer)
             }
         }
 
-        private class TwoPassTranscoderHelper<FPW> : TranscoderHelper
+        class TwoPassTranscoderHelper<FPW> : TranscoderHelper
         {
-            readonly Action<R, FPW>[] firstPassTranscode;
+            readonly Lazy<Action<R, FPW>[]> firstPassTranscode;
 
             public TwoPassTranscoderHelper(RuntimeSchema schema, IParser parser):
-                base(schema, null)
+                base(schema, parser)
```

Comment:
This change looks like it's addressing a bug. Did the unit tests catch the bug before this fix? If not, we'll want to add a regression test.